users/raissi/setups/common/decoder/BASE_factored_hybrid_search.py

```diff
@@ -917,13 +919,30 @@ def recognize(
             is_count_based=True,
         )
 
+        adv_search_extra_config = (
+            copy.deepcopy(adv_search_extra_config)
+            if adv_search_extra_config is not None
+            else rasr.RasrConfig()
+        )
         if search_parameters.altas is not None:
-            adv_search_extra_config = rasr.RasrConfig()
             adv_search_extra_config.flf_lattice_tool.network.recognizer.recognizer.acoustic_lookahead_temporal_approximation_scale = (
                 search_parameters.altas
             )
-        else:
-            adv_search_extra_config = None
+        if search_parameters.lm_lookahead_scale is not None:
+            # Use 4gram for lookahead. The lookahead LM must not be too good.
+            #
+            # Half the normal LM scale is a good value here.
+
+            adv_search_extra_config.flf_lattice_tool.network.recognizer.recognizer.separate_lookahead_lm = True
+            adv_search_extra_config.flf_lattice_tool.network.recognizer.recognizer.lm_lookahead.lm_lookahead_scale = (
+                search_parameters.lm_lookahead_scale
+            )
+            adv_search_extra_config.flf_lattice_tool.network.recognizer.recognizer.lookahead_lm.file = (
+                original_lm_config.language_model_config.file
+            )
+            # TODO(future): Add LM image instead of file here.
+            adv_search_extra_config.flf_lattice_tool.network.recognizer.recognizer.lookahead_lm.scale = 1.0
+            adv_search_extra_config.flf_lattice_tool.network.recognizer.recognizer.lookahead_lm.type = "ARPA"
```

Comment:
So you confirm that the "real" lm lookahead scale in this case is under `lm_lookahead.lookahead_scale` and not here?

Reply:
I believe the lm_lookahead_scale scales the lookahead during decoding, using LM scores scaled by the `lookahead_lm.scale`. I'm not entirely sure with that though, but from testing I discovered the scales do not have the same effect. For 4gram lookahead this configuration works well.

Reply:
I don't know by heart, this is unfortunately quite complicated due to historic reasons. I'd need to look into this. Maybe @SimBe195 remembers.

Reply:
Yeah there was some very confusing logic with these scales. I looked at this a while ago but can't remember the details. I believe this was even handled a bit inconsistently in the code, like sometimes the `lookahead-lm.scale` and `lm-lookahead-scale` would be multiplied together and sometimes only the lm-lookahead-scale would be applied. It gets even more confusing when you have a `CombineLM` with sub-lm's that also have their own scale in addition.